test/e2e/filters/http_headers_modifier/main.go

```diff
@@ -6,22 +6,32 @@ import (
 	"github.com/ardikabs/gonvoy"
 )
 
-func init() {
-	gonvoy.RunHttpFilter(new(Filter), gonvoy.ConfigOptions{
-		FilterConfig: new(Config),
-	})
+const filterName = "http_headers_modifier"
 
-	gonvoy.RunHttpFilter(new(Echoserver), gonvoy.ConfigOptions{})
+func init() {
+	gonvoy.RunHttpFilter(
+		filterName,
+		func() gonvoy.HttpFilter {
+			return new(Filter)
+		},
+		gonvoy.ConfigOptions{
+			FilterConfig: new(Config),
+		},
+	)
+
+	gonvoy.RunHttpFilter(
+		filterName,
```

Comment:
this e2e tests use 2 different filters, the `http_headers_modifier` and `echoserver`, hence it shouldn't reuse the `filterName`.

Reply:
Sounds good to me. I will update the test files when back at home today or tomorrow.